backend/apps/api/src/routes/image.py

```
@@ -0,0 +1,56 @@
+):
+    """Presigned URL 생성 (다운로드/조회용)"""
+    if key.startswith("/") or ".." in key:
+        raise HTTPException(status_code=400, detail="Invalid key")
+
+    # 파일 존재 확인
+    try:
+        s3_internal.head_object(Bucket=S3_BUCKET, Key=key)
+    except ClientError as e:
+        code = e.response.get("Error", {}).get("Code")
+        if code in ("404", "NoSuchKey", "NotFound"):
+            raise HTTPException(status_code=404, detail="Object not found")
```

Ruff (0.14.1)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)